scylla/src/errors.rs

```diff
@@ -362,32 +385,60 @@ pub struct ConnectionSetupRequestError {
 #[non_exhaustive]
 pub enum ConnectionSetupRequestErrorKind {
     // TODO: Make FrameError clonable.
+    /// An error occurred when parsing response frame header.
     #[error(transparent)]
     FrameError(Arc<FrameError>),
+
+    /// Driver was unable to allocate a stream id to execute a setup request on.
     #[error("Unable to allocate stream id")]
     UnableToAllocStreamId,
+
+    /// A connection was broken during setup request execution.
     #[error(transparent)]
     BrokenConnection(#[from] BrokenConnectionError),
+
+    /// Received a server error in response to connection setup request.
     #[error("Database returned an error: {0}, Error message: {1}")]
     DbError(DbError, String),
+
+    /// Received an unexpected response from the server.
     #[error("Received unexpected response from the server: {0}")]
     UnexpectedResponse(CqlResponseKind),
+
+    /// Received a response to OPTIONS request, but failed to deserialize its body.
     #[error("Failed to deserialize SUPPORTED response: {0}")]
     CqlSupportedParseError(#[from] CqlSupportedParseError),
+
+    /// Received an AUTHENTICATE response, but failed to deserialize its body.
     #[error("Failed to deserialize AUTHENTICATE response: {0}")]
     CqlAuthenticateParseError(#[from] CqlAuthenticateParseError),
+
+    /// Received an AUTH_SUCCESS response, but failed to deserialize its body.
     #[error("Failed to deserialize AUTH_SUCCESS response: {0}")]
     CqlAuthSuccessParseError(#[from] CqlAuthSuccessParseError),
+
+    /// Received an AUTH_CHALLENGE response, but failed to deserialize its body.
     #[error("Failed to deserialize AUTH_CHALLENGE response: {0}")]
     CqlAuthChallengeParseError(#[from] CqlAuthChallengeParseError),
+
+    /// Received server ERROR response, but failed to deserialize its body.
     #[error("Failed to deserialize ERROR response: {0}")]
     CqlErrorParseError(#[from] CqlErrorParseError),
+
+    /// An error returned by [`AuthenticatorProvider::start_authentication_session`](crate::authentication::AuthenticatorProvider::start_authentication_session).
     #[error("Failed to start client's auth session: {0}")]
     StartAuthSessionError(AuthError),
+
+    /// An error returned by [`AuthenticatorSession::evaluate_challenge`](crate::authentication::AuthenticatorSession::evaluate_challenge).
     #[error("Failed to evaluate auth challenge on client side: {0}")]
     AuthChallengeEvaluationError(AuthError),
+
+    /// An error returned by [`AuthenticatorSession::success`](crate::authentication::AuthenticatorSession::success).
     #[error("Failed to finish auth challenge on client side: {0}")]
     AuthFinishError(AuthError),
+
+    /// User did not provide authentication. See [`SessionBuilder::user`](crate::transport::session_builder::SessionBuilder::user)
+    /// and/or [`SessionBuilder::authenticator_provider`](crate::transport::session_builder::SessionBuilder::authenticator_provider).
```

Comment:
Suggested clarification:
`User did not provide authentication.` -> `User did not provide authentication while the cluster requires it.`